docs/src/content/docs/getting-started/learning.md

```diff
@@ -0,0 +1,177 @@
+```typescript
+const accAtom = atom<number>((ctx, prev = 0) => prev + 1, 'accAtom')
+```
+
+
+
+## Actions enable transactions
+Setting atoms manually is good thing but more often we want to do many changes at once.
+Let's edit example so that we are able to change `a` and `b` simultaniously.
+
+```typescript
+import { createCtx, atom, action } from "@reatom/framework"
+
+const ctx = createCtx()
+
+const aAtom = atom(1, 'aAtom')
+const bAtom = atom(2, 'bAtom')
+
+const setParams = action((ctx, a: number, b: number) => {
+  console.log(`change a=${a}, b=${b}`)
+  aAtom(ctx, a)
+  bAtom(ctx, b)
+}, 'setParams')
+
+const cAtom = atom((ctx) =>
+  ctx.spy(aAtom) + ctx.spy(bAtom),
+  'cAtom'
+)
+
+ctx.subscribe(cAtom, (c) => {
+  const a = ctx.get(aAtom);
+  const b = ctx.get(bAtom);
+  console.log(`${a} + ${b} = ${c}`)
+})
+
+setParams(ctx, 10, 12)
+// logs: change a=10, b=12
+// logs: 10 + 12 = 22
+
+setParams(ctx, 10, 12)
+// logs: change a=10, b=12
+// does not log the result since it did not change
+
+bAtom(ctx, 4)
+// logs: 10 + 4 = 14
+```
+
+As we see here the subscribe callback only called with both a and b values changed.
+This is called transactions and they enable us to control how frequent consumer of state is notified of some changes.
+
+<!--
+
+## Schedule to use asyncronous actions
+Sometimes you need to call something to be able to change something. Reatom does not support asyncronous transactions yet.
+Instead there's a special mechanism of queues that enble us to do something.
+
+TODO example with ctx.schedule and ctx.schedule -1
+
+You can take a deeper dive in queues here
+
+-->
+
+## Contexts define different universes
+Contexts are used to glue up atoms and actions, track transactions and many more features. You can use same dependency trees in different contexts:
+```typescript
+import { createCtx, atom } from "@reatom/framework"
+
+const ctx1 = createCtx()
+const ctx2 = createCtx()
+
+cons someAtom = atom(1, 'someAtom')
+
+console.log(ctx1.get(someAtom), ctx2.get(someAtom))
+// logs: 1, 1
+
+// change value of an atom only in one context
+someAtom(ctx1, 100)
+
+console.log(ctx1.get(someAtom), ctx2.get(someAtom))
+// logs: 100, 1
+```
+
+Computed atoms can't depend on atom values from different contexts and actions can't change atoms of different context.
+Context will initiate new item state referring to that atom.
+
+This enables us to easily test things. But beware of function closures because they are not context dependent!
+
+```typescript
+import { createCtx, atom } from "@reatom/framework"
+
+const ctx1 = createCtx()
+const ctx2 = createCtx()
+
+// DON'T DO THIS
+let someExternalVar = 1
+cons someAtom = atom(() => someExternalVar, 'someAtom')
+
+// check using ctx1
+console.log(ctx1.get(someAtom))
+// logs: 1
+
+someExternalVar = 100
+
+// check using ctx1 and ctx2
+console.log(ctx1.get(someAtom), ctx2.get(someAtom))
+// logs: 1, 100
+// because ctx1 cached 1 and ctx2 was only read when value changed
+```
```

Comment:
```suggestion
```

To fix this, simply wrap the value in atom and spy, and change it within the relative context.
```